bftengine/src/bcstatetransfer/Messages.hpp

```diff
@@ -66,11 +66,18 @@ struct CheckpointSummaryMsg : public BCStateTranBaseMsg {
   uint64_t requestMsgSeqNum;
 
   static bool equivalent(const CheckpointSummaryMsg* a, const CheckpointSummaryMsg* b) {
-    return (memcmp(a, b, sizeof(CheckpointSummaryMsg)) == 0);
+    return (a->lastBlock == b->lastBlock and
+      a->checkpointNum == b->checkpointNum and
+      (memcmp(&a->digestOfLastBlock, &b->digestOfLastBlock, sizeof(STDigest)) == 0) and
+      (memcmp(&a->digestOfResPagesDescriptor, &b->digestOfResPagesDescriptor, sizeof(STDigest)) == 0));
   }
```

Comment:
I see that for some expressions you put brackets and some not. I recommend having brackets around all. same for the next change.